salt/_states/metalk8s.py

```diff
@@ -0,0 +1,32 @@
+"""Custom states for MetalK8s."""
+
+__virtualname__ = "metalk8s"
+
+
+def __virtual__():
+    return __virtualname__
+
+
+def static_pod_managed(name, source, config_files, context):
+    """Simple helper to edit a static Pod manifest if configuration changes.
+
+    Expects the template to use the `config_digests` variable and store it in
+    the `metadata.annotations` section, with the key
+    `metalk8s.scality.com/config-digests`.
+    """
+    config_digests = [
+        __salt__["hashutil.digest_file"](config_file, checksum="sha256")
+        for config_file in config_files
+    ]
+
+    return __states__["file.managed"](
+        name,
+        source,
+        template="jinja",
+        user="root",
+        group="root",
+        mode="0644",
+        makedirs=False,
+        backup=False,
+        context=dict(context, config_digests=",".join(config_digests)),
```

Comment:
Uh, nah. That'd mean we can get some very long but otherwise useless annotation on the Pod. If indeed we want to use `sha256` (which isn't even required here), then the result should be, I don't know, a checksum of the concatenation of all sorted checksums or something? Basically, one single value, ideally short, which is (quasi-)guaranteed to change when the input(s) change.

Reply:
Would something along the lines of `md5( '-'.join( sha256(file) for file in config_files )` suit our needs ?

Reply:
Done